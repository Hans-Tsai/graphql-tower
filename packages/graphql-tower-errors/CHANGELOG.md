# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="3.21.3"></a>
## [3.21.3](https://github.com/tmotx/graphql-tower/compare/v3.21.2...v3.21.3) (2017-12-05)




**Note:** Version bump only for package graphql-tower-errors

<a name="3.16.0"></a>
# [3.16.0](https://github.com/tmotx/graphql-tower/compare/v3.15.1...v3.16.0) (2017-11-19)


### Features

* **help:** added assertResult ([e6cd019](https://github.com/tmotx/graphql-tower/commit/e6cd019))




<a name="3.15.1"></a>
## [3.15.1](https://github.com/tmotx/graphql-tower/compare/v3.15.0...v3.15.1) (2017-11-18)


### Bug Fixes

* **error:** added NotAcceptableError ([7258d43](https://github.com/tmotx/graphql-tower/commit/7258d43))
