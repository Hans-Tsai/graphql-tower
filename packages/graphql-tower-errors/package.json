{
  "name": "graphql-tower-errors",
  "version": "1.0.0",
  "description": "",
  "main": "lib/index.js",
  "scripts": {
    "build": "rm -rf lib/* && babel src --ignore __tests__ --out-dir lib"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tmotx/graphql-tower.git"
  },
  "author": "tmotx",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tmotx/graphql-tower/issues"
  },
  "homepage": "https://github.com/tmotx/graphql-tower#readme",
  "files": [
    "lib"
  ],
  "peerDependencies": {
    "graphql": "^0.11.7"
  },
  "devDependencies": {
    "faker": "^4.1.0"
  }
}
