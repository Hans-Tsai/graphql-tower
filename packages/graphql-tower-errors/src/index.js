import { GraphQLError } from 'graphql/error';

export class ForbiddenError extends GraphQLError {
  symbol = 'ForbiddenError';

  constructor(message = 'ForbiddenError') {
    super(message);
  }
}

export class NotFoundError extends GraphQLError {
  symbol = 'NotFoundError';

  constructor(message = 'NotFoundError') {
    super(message);
  }
}

export class UnauthorizedError extends GraphQLError {
  symbol = 'UnauthorizedError';

  constructor(message = 'UnauthorizedError') {
    super(message);
  }
}

export class PaymentRequiredError extends GraphQLError {
  symbol = 'PaymentRequiredError';

  constructor(message = 'PaymentRequiredError') {
    super(message);
  }
}

export class NotAcceptableError extends GraphQLError {
  symbol = 'NotAcceptableError';

  constructor(message = 'NotAcceptableError') {
    super(message);
  }
}

export class GoneDataError extends GraphQLError {
  symbol = 'GoneDataError';

  constructor(message = 'GoneDataError') {
    super(message);
  }
}

export class ConflictError extends GraphQLError {
  symbol = 'ConflictError';

  constructor(message = 'ConflictError') {
    super(message);
  }
}

export class LockedError extends GraphQLError {
  symbol = 'LockedError';

  constructor(message = 'LockedError') {
    super(message);
  }
}

export class UnavailableForLegalReasonsError extends GraphQLError {
  symbol = 'UnavailableForLegalReasonsError';

  constructor(message = 'UnavailableForLegalReasonsError') {
    super(message);
  }
}
