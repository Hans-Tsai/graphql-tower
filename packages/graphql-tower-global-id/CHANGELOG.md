# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="3.39.10"></a>
## [3.39.10](https://github.com/tmotx/graphql-tower/compare/v3.39.9...v3.39.10) (2018-08-15)


### Bug Fixes

* **all:** update packages ([e37ac1a](https://github.com/tmotx/graphql-tower/commit/e37ac1a))





<a name="3.3.3"></a>
## [3.3.3](https://github.com/tmotx/graphql-tower/compare/v3.3.2...v3.3.3) (2017-10-27)


### Bug Fixes

* **cache:** fix get model from cache will load model not data ([d7903b7](https://github.com/tmotx/graphql-tower/commit/d7903b7))




<a name="3.1.0"></a>
# [3.1.0](https://github.com/tmotx/graphql-tower/compare/v3.0.1...v3.1.0) (2017-10-26)


### Bug Fixes

* **tower:** fix export default ([12f42ca](https://github.com/tmotx/graphql-tower/commit/12f42ca))
