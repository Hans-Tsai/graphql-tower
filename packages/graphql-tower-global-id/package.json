{
  "name": "graphql-tower-global-id",
  "version": "1.0.0",
  "description": "",
  "main": "lib/index.js",
  "scripts": {
    "build": "rm -rf lib/* && babel src --ignore __tests__ --out-dir lib"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tmotx/graphql-tower.git"
  },
  "author": "tmotx",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tmotx/graphql-tower/issues"
  },
  "homepage": "https://github.com/tmotx/graphql-tower#readme",
  "files": [
    "lib"
  ],
  "dependencies": {
    "base-x": "^3.0.2",
    "bignumber.js": "^4.1.0",
    "lodash": "^4.17.4"
  }
}
