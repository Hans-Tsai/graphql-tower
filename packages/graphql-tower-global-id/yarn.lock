# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


base-x@^3.0.2:
  version "3.0.2"
  resolved "https://registry.yarnpkg.com/base-x/-/base-x-3.0.2.tgz#bf873861b7514279b7969f340929eab87c11d130"
  dependencies:
    safe-buffer "^5.0.1"

bignumber.js@^4.1.0:
  version "4.1.0"
  resolved "https://registry.yarnpkg.com/bignumber.js/-/bignumber.js-4.1.0.tgz#db6f14067c140bd46624815a7916c92d9b6c24b1"

lodash@^4.17.4:
  version "4.17.4"
  resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.4.tgz#78203a4d1c328ae1d86dca6460e369b57f4055ae"

safe-buffer@^5.0.1:
  version "5.1.1"
  resolved "https://registry.yarnpkg.com/safe-buffer/-/safe-buffer-5.1.1.tgz#893312af69b2123def71f57889001671eeb2c853"
