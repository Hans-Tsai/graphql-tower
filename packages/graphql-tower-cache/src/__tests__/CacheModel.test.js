import _ from 'lodash';
import knex, { client } from 'jest-mock-knex';
import Model, { ValueColumn } from 'graphql-tower-model';
import { toGlobalId } from 'graphql-tower-global-id';
import { NotFoundError } from 'graphql-tower-errors';
import { CacheModel } from '../';

process.setMaxListeners(0);

const database = knex({
  client: 'pg',
  connection: {
    host: '127.0.0.1',
    user: 'postgres',
    password: null,
    database: 'graphql_tower',
  },
});

class CarModel extends Model {
  static database = database;

  static tableName = 'car';

  static columns = {
    name: new ValueColumn(),
  }
}

class HouseModel extends Model {
  static database = database;

  static tableName = 'house';

  static columns = {
    name: new ValueColumn(),
  }
}

class Cache extends CacheModel {
  static car = CarModel;
  static house = HouseModel;
}

describe('CacheModel', () => {
  afterAll(() => database.destroy());

  describe('load & loadMany', () => {
    it('cache 1', async () => {
      const cache = new Cache();

      client.mockReturnValueOnce([{ id: '1', name: '1 of car' }]);
      await cache.load(toGlobalId('car', '1'));

      client.mockReturnValueOnce([{ id: '1', name: '1 of house' }]);
      await cache.load(toGlobalId('house', '1'));

      client.mockReturnValueOnce([{ id: '3', name: '3 of car' }]);
      client.mockReturnValueOnce([{ id: '2', name: '2 of house' }, { id: '4', name: '4 of house' }]);

      await Promise.all([
        cache.load(toGlobalId('car', '1')),
        cache.load(toGlobalId('house', '2')),
        cache.loadMany([toGlobalId('car', '3'), toGlobalId('house', '1'), toGlobalId('house', '4')]),
      ]);

      expect(client).toMatchSnapshot();
      expect(client).toHaveBeenCalledTimes(4);
    });

    it('cache 2', async () => {
      const cache = new Cache();

      client.mockReturnValueOnce([{ id: '1', name: '1 of car' }]);
      await cache.load(toGlobalId('car', '1'));

      client.mockReturnValueOnce([{ id: '1', name: '1 of house' }]);
      await cache.load(toGlobalId('house', '1'));

      expect(client).toMatchSnapshot();
      expect(client).toHaveBeenCalledTimes(2);
    });
  });

  it('check type', async () => {
    const cache = new Cache();

    await expect(cache.load(toGlobalId('tree', '1'))).resolves.toBeNull();

    client.mockReturnValueOnce([{ id: '1', name: '1 of car' }]);
    await expect(cache.load(toGlobalId('car', '1'), 'house')).resolves.toBeNull();

    await expect(cache.load(toGlobalId('car', '1'), 'house', NotFoundError))
      .rejects.toEqual(new NotFoundError());

    expect(client).toMatchSnapshot();
    expect(client).toHaveBeenCalledTimes(1);
  });

  it('prime', async () => {
    const cache = new Cache();
    cache.prime(toGlobalId('car', '1'), { id: '1', name: '1 of car' });

    await expect(cache.load(toGlobalId('car', '1')))
      .resolves.toEqual(expect.objectContaining({ id: toGlobalId('car', '1'), name: '1 of car' }));
  });

  it('clear & clearAll', async () => {
    const cache = new Cache();

    client.mockReturnValueOnce([{ id: '1', name: '1 of car' }]);
    client.mockReturnValueOnce([{ id: '1', name: '1 of car' }]);
    client.mockReturnValueOnce([{ id: '1', name: '1 of car' }]);

    await cache.load(toGlobalId('car', '1'));
    await cache.load(toGlobalId('car', '1'));
    cache.clear(toGlobalId('car', '1'));
    await cache.load(toGlobalId('car', '1'));
    cache.clearAll();
    await cache.load(toGlobalId('car', '1'));

    expect(client).toMatchSnapshot();
    expect(client).toHaveBeenCalledTimes(3);
  });

  it('new model', async () => {
    const cache = new Cache();
    const { car } = cache;

    client.mockReturnValueOnce([{ id: '1', name: '1 of car' }, { id: '2', name: '2 of car' }]);
    expect(_.size(await car.fetchAll())).toBe(2);

    await cache.load(toGlobalId('car', '1'));

    expect(client).toMatchSnapshot();
    expect(client).toHaveBeenCalledTimes(1);
  });
});
