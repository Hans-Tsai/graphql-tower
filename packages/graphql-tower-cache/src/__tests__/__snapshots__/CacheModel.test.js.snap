// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CacheModel check type 1`] = `
Array [
  "select * from car where id in (1) and deleted_at is null",
]
`;

exports[`CacheModel clear & clearAll 1`] = `
Array [
  "select * from car where id in (1) and deleted_at is null",
  "select * from car where id in (1) and deleted_at is null",
  "select * from car where id in (1) and deleted_at is null",
]
`;

exports[`CacheModel load & loadMany cache 1 1`] = `
Array [
  "select * from car where id in (1) and deleted_at is null",
  "select * from house where id in (1) and deleted_at is null",
  "select * from house where id in (2, 4) and deleted_at is null",
  "select * from car where id in (3) and deleted_at is null",
]
`;

exports[`CacheModel load & loadMany cache 2 1`] = `
Array [
  "select * from car where id in (1) and deleted_at is null",
  "select * from house where id in (1) and deleted_at is null",
]
`;

exports[`CacheModel new model 1`] = `
Array [
  "select * from car where deleted_at is null limit 1000",
]
`;
